internal/driver/gl/gl.go

```diff
@@ -248,6 +248,15 @@ func (c *glCanvas) newGlRasterTexture(obj fyne.CanvasObject) uint32 {
 	return c.imgToTexture(rast.Generator(width, height))
 }
 
+func (c *glCanvas) newGlGradientTexture(obj fyne.CanvasObject) uint32 {
+	rast := obj.(*canvas.Gradient)
+
+	width := textureScaleInt(c, rast.Size().Width)
+	height := textureScaleInt(c, rast.Size().Height)
+
+	return c.imgToTexture(rast.Generator(width, height))
```

Comment:
I didn't realise until now that this is literally a raster with a different generator function. The idea that Cedric and I had about potential hardware acceleration is not possible if the details of the gradient (colors, direction etc) are hidden within the type. Perhaps we can discuss if it's possible to declare more as state and render using some common code that can be hardware optimised later?